src/imagery/i.sam2/i.sam2.py

```diff
@@ -75,116 +74,190 @@
+            box_threshold=box_threshold,
+            text_threshold=text_threshold,
+            return_results=True,
         )
-        rgb_array = rgb_array.astype(np.uint8)
+    return masks
 
-    np_image = Image.fromarray(rgb_array[:, :, :3])
 
-    # Get device
-    device = "cuda" if torch.cuda.is_available() else "cpu"
-    if device == "cuda":
-        torch.cuda.empty_cache()
+def run_samgeo_segmentation(temp_input_path, temp_output_path, model_path, device):
+    """
+    Run SAMGeo segmentation on an input image and save the output.
 
-    try:
-        if text_prompt:
-            gs.message("Running LangSAM segmentation...")
-            sam = LangSAM(
-                model_type="sam2-hiera-large",
-            )
-            from torch.amp.autocast_mode import autocast
-
-            with autocast(device_type=device):
-                masks, boxes, phrases, logits = sam.predict(
-                    image=np_image,
-                    text_prompt=text_prompt,
-                    box_threshold=box_threshold,
-                    text_threshold=text_threshold,
-                    return_results=True,
-                )
-        else:
-            gs.message("Running SAMGeo segmentation...")
-            sam = SamGeo(model_type="vit_h", model_path=model_path, device=device)
-            sam.generate(
-                input_image=temp_input_path,
-                output=temp_output_path,
-                guide=guide_input_path,
-            )
-    except Exception as e:
-        gs.message(torch.cuda.memory_summary())
-        gs.fatal(f"Error while running SAMGeo: {e}")
-        return 1
+    Parameters:
+    temp_input_path (str): The file path to the input image.
+    temp_output_path (str): The file path to save the segmented output image.
+    model_path (str): The file path to the SAMGeo model.
+    device (str): The device to run the model on (e.g., 'cpu', 'cuda').
 
-    gs.message("Segmentation complete.")
+    Returns:
+    None
+    """
+    from samgeo import SamGeo
 
-    write_raster(masks, output_raster)
-    return 0
+    gs.message(_("Running SAMGeo segmentation..."))
+    sam = SamGeo(model_type="vit_h", model_path=model_path, device=device)
+    sam.generate(input_image=temp_input_path, output=temp_output_path)
 
 
 def write_raster(input_np_array, output_raster):
-    gs.message("Importing the segmented raster into GRASS GIS...")
+    """
+    Writes a segmented raster into GRASS GIS.
+
+    Parameters:
+    input_np_array (list of numpy.ndarray): A list of numpy arrays representing the input raster bands.
+    output_raster (str): The name of the output raster map to be created in GRASS GIS.
+
+    Raises:
+    ValueError: If the input array is empty or if the masks do not have the same shape.
+
+    This function merges multiple raster bands into a single raster, where each band is assigned a unique value.
+    The merged raster is then written to a GRASS GIS raster map.
```

Comment:
This is the first time you use raster "bands". Would it be equivalent to use raster maps or segmented masks?

Reply:
Yes. I will switch it should say "input raster masks".